galley/pkg/config/analysis/analyzers/gateway/gateway.go

```diff
@@ -86,7 +86,7 @@ func (*IngressGatewayPortAnalyzer) analyzeGateway(r *resource.Instance, c analys
 				if svcSelector.Matches(podLabels) {
 					for _, port := range service.Ports {
 						if port.Protocol == "TCP" {
-							servicePorts[uint32(port.Port)] = true
+							servicePorts[uint32(port.TargetPort.IntValue())] = true
```

Comment:
I don't think this is sufficient as ports `80` and `443` are special cased for Gateway resources. If you make this change then all GW resources will port 80 and 443 will start to emit incorrect warning message.

Reply:
80 and 443 are not special cases. envoy will listen on the port provided in the gateway, unless the associated service has it as a targetPort, in which case it will listen on the service port.

I do still think this logic isn't quite right though

Reply:
> envoy will listen on the port provided in the gateway, unless the associated service has it as a targetPort


This PR checks `port.Protocol == "TCP"`  which is the current defaults for `ingressgateway` and do have `targetPort`.
https://github.com/istio/istio/blob/42a8c73c10cec19cb8ea28a436a1eeab1952d5a1/manifests/charts/gateways/istio-ingress/values.yaml#L15-L31

> envoy will listen on the port provided in the gateway.

Not sure what will happen if different port is provided in gateway?

Reply:
It does not matter if our defaults have targetPort, we need the analyzer to work for non-default cases.

If I have a gateway with ports 1000,2000,3000, 4000
Service with

port 1000->target 1001
port 2000 no target port
port 3000 -> target 3000
port 5000 no target port

I will get envoy listeners on: 1001, 2000, 3000, 4000